dev/release/verify-release-candidate.sh

```diff
@@ -591,7 +591,7 @@ test_and_install_cpp() {
     -DPARQUET_REQUIRE_ENCRYPTION=ON \
     ${ARROW_CMAKE_OPTIONS:-} \
     ${ARROW_SOURCE_DIR}/cpp
-  export CMAKE_BUILD_PARALLEL_LEVEL=${CMAKE_BUILD_PARALLEL_LEVEL:-${NPROC}}
+  export CMAKE_BUILD_PARALLEL_LEVEL=${ARROW_BUILD_PARALLEL:-${NPROC}}
```

Comment:
Hmm. I think that we don't need this because we always use CMake in this script.
If we want this, could you use use `ARROW_BUILD_PARALLEL` only if `CMAKE_BUILD_PARALLEL` isn't defined?

Reply:
Good point - for now I have just reverted this